src/js/modal-maintenance.js

```diff
@@ -0,0 +1,253 @@
+'use strict';
+
+import {broadcast} from 'n-ui-foundations';
+import oViewport from 'o-viewport';
+import Superstore from 'superstore';
+
+import {TRACKING} from './config';
+
+import {toElement} from './util';
+import {getAllItemsForID, getItemByHTMLElement} from './data-store';
+
+const localStore = new Superstore('local', 'syndication');
+
+let OVERLAY_FRAGMENT;
+let OVERLAY_MODAL_ELEMENT;
+let OVERLAY_SHADOW_ELEMENT;
+let USER_DATA;
+let DAYS_LEFT;
+
+function init (user, daysLeft) {
+	addEventListener('click', actionModalFromClick, true);
+
+	addEventListener('keyup', actionModalFromKeyboard, true);
+	addEventListener('resize', reposition, true);
+
+	oViewport.listenTo('resize');
+
+	USER_DATA = user;
+	DAYS_LEFT = daysLeft;
+}
+
+function daysUntilMaintenance (date) {
+	let dateNow = new Date(Date.now());
+	let maintenanceDate = new Date(date);
+	const diffTime = maintenanceDate - dateNow;
+	const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+
+	if (diffDays>=0){
+		return diffDays;
+	}
+	return -1;
+}
+
+function actionModalFromClick (evt) {
```

Comment:
Normally I'd request to spell out "event" as we try to avoid abbreviations, but I can see the repo already uses "evt" so probably best to keep as is to be consistent within the repo

Reply:
sounds good. no changes needed